filament/src/ShadowMap.cpp

```diff
@@ -103,9 +103,47 @@ void ShadowMap::render(DriverApi& driver, Handle<HwRenderTarget> rt,
     pass.execute("Shadow map Pass", rt, params);
 }
 
+void ShadowMap::computeSceneCascadeParams(const FScene::LightSoa& lightData, size_t index,
+        FScene const* scene, details::CameraInfo const& camera, uint8_t visibleLayers,
+        CascadeParameters& cascadeParams) {
+    // Compute the light's model matrix.
+    const float3 lightPosition = camera.getPosition();
+    const float3 dir = lightData.elementAt<FScene::DIRECTION>(index);
+    const mat4f M = mat4f::lookAt(lightPosition, lightPosition + dir, float3{ 0, 1, 0 });
+    const mat4f Mv = FCamera::rigidTransformInverse(M);
+    const mat4f V = camera.view;
+
+    float2& lsNearFar = cascadeParams.lsNearFar;
+    float2& vsNearFar = cascadeParams.vsNearFar;
+    Aabb& wsShadowCastersVolume = cascadeParams.wsShadowCastersVolume;
+    Aabb& wsShadowReceiversVolume = cascadeParams.wsShadowReceiversVolume;
```

Comment:
instead of using references here, just use `cascadeParams` everywhere, including in the lambda below -- it'll be less confusing and more efficient.